data/gui/themes/default/widget/window_transparent.cfg

```diff
@@ -0,0 +1,93 @@
+#textdomain wesnoth-lib
+###
+### Definition of transparent dialogs.
+###
+
+#define _GUI_RESOLUTION_TRANSPARENT_BASE RESOLUTION BORDER_SIDE_1 BORDER_SIDE_2 _BACKGROUND_DRAW
+	[resolution]
+
+		{RESOLUTION}
+
+		left_border = {BORDER_SIDE_1}
+		right_border = {BORDER_SIDE_2}
+		top_border = {BORDER_SIDE_1}
+		bottom_border = {BORDER_SIDE_2}
+
+		[background]
+
+			[draw]
+
+				{_BACKGROUND_DRAW}
```

Comment:
Please don't use the underscore.
The convention is to start any macros defined in GUI WML with _GUI, but BACKGROUND_DRAW is a macro argument.

I'm guessing this argument is for the BACKGROUND image to be shown?

Reply:
Just remove that argument entirely.

Reply:
> Just remove that argument entirely.

That's also another way. The person using it could show their image using a `[image]` or `[drawing]` widget. Otherwise they'll have to create a new definition for each different background.

Reply:
I mean, I honestly think that would be a better solution than this. You can already create custom window definitions, so it would be better to just define the background there instead of having this "transparent" definition.

Reply:
Having a transparent window is likely easier for people making dialogs in Lua though.

Reply:
Tbh, I amn't quite sure why RESOLUTION and _BACKGROUND_DRAW arguments are even there, but same things can be found in window_default.cfg and window_borderless.cfg, so I decided not to change anything here, in case it's important and may break something. If they are not needed and do not perform any functions, which is the most likely the case, I think they can be removed here and from other files.

Overall, I think the addition of window_transparent is a good thing, especially for beginners who will find it difficult to understand very confusing and small GUI documentation. But if there are better options to add more customization to dialogs without defining new windows, it would be very good

Reply:
Is it difficult to add a new window definition from lua?

Reply:
Well, yes. I couldn't figure it out, therefore I thought that other new add-on makers might face a similar problem. Of course, I may be just stupid, but I think that wiki doesn't give enough information and there are almost no working examples
![Screenshot_1397](https://github.com/user-attachments/assets/3bca6ff0-9b4b-4e17-a746-be81061420e8)